File: TeamCode/src/main/java/pedroPathing/constants/FConstants.java
package pedroPathing.constants;

import com.pedropathing.follower.FollowerConstants;
import com.pedropathing.localization.Localizers;
import com.pedropathing.util.KalmanFilterParameters;
import com.qualcomm.robotcore.hardware.DcMotorSimple;

public class FConstants {
    static {
        FollowerConstants.localizers = Localizers.THREE_WHEEL_IMU;

        FollowerConstants.leftFrontMotorName = "MotorFrontLeft";
        FollowerConstants.leftRearMotorName = "MotorBackLeft";
        FollowerConstants.rightFrontMotorName = "MotorFrontRight";
        FollowerConstants.rightRearMotorName = "MotorBackRight";

        FollowerConstants.leftFrontMotorDirection = DcMotorSimple.Direction.REVERSE;
        FollowerConstants.leftRearMotorDirection = DcMotorSimple.Direction.REVERSE;
        FollowerConstants.rightFrontMotorDirection = DcMotorSimple.Direction.FORWARD;
        FollowerConstants.rightRearMotorDirection = DcMotorSimple.Direction.FORWARD;

        FollowerConstants.mass = 14.5;

        FollowerConstants.xMovement = 70.23046066410981;
        FollowerConstants.yMovement = 40.37395651785481;

        FollowerConstants.forwardZeroPowerAcceleration = -41.78295044292578;
        FollowerConstants.lateralZeroPowerAcceleration = -73.40248127111744;

        FollowerConstants.translationalPIDFCoefficients.setCoefficients(0.3,0,0.016,0);
        FollowerConstants.useSecondaryTranslationalPID = false;
        FollowerConstants.secondaryTranslationalPIDFCoefficients.setCoefficients(0.1,0,0.01,0); // Not being used, @see useSecondaryTranslationalPID

        FollowerConstants.headingPIDFCoefficients.setCoefficients(1.5,0,0.15,0);
        FollowerConstants.useSecondaryHeadingPID = false;
        FollowerConstants.secondaryHeadingPIDFCoefficients.setCoefficients(2,0,0.1,0); // Not being used, @see useSecondaryHeadingPID

        FollowerConstants.drivePIDFCoefficients.setCoefficients(0.0095,0,0.0019,0.5,0);
        FollowerConstants.useSecondaryDrivePID = false;
        FollowerConstants.secondaryDrivePIDFCoefficients.setCoefficients(0.1,0,0,0.6,0); // Not being used, @see useSecondaryDrivePID

        FollowerConstants.driveKalmanFilterParameters = new KalmanFilterParameters(7, 1.2);

        FollowerConstants.zeroPowerAccelerationMultiplier = 4;
        FollowerConstants.centripetalScaling = .0005;

        FollowerConstants.pathEndTimeoutConstraint = 300;
        FollowerConstants.pathEndTValueConstraint = 0.995;
        FollowerConstants.pathEndVelocityConstraint = 0.1;
        FollowerConstants.pathEndTranslationalConstraint = 0.1;
        FollowerConstants.pathEndHeadingConstraint = 0.005;

//        FollowerConstants.automaticHoldEnd = false;
    }
}

File: TeamCode/src/main/java/pedroPathing/constants/FConstants.kt
package pedroPathing.constants

import com.pedropathing.follower.FollowerConstants
import com.pedropathing.localization.Localizers
import com.qualcomm.robotcore.hardware.DcMotorSimple

@Suppress("unused")
object FConstants {
    init {
        FollowerConstants.localizers = Localizers.THREE_WHEEL

        FollowerConstants.leftFrontMotorName = "MotorFrontLeft"
        FollowerConstants.leftRearMotorName = "MotorBackLeft"
        FollowerConstants.rightFrontMotorName = "MotorFrontRight"
        FollowerConstants.rightRearMotorName = "MotorBackRight"

        FollowerConstants.leftFrontMotorDirection = DcMotorSimple.Direction.REVERSE
        FollowerConstants.leftRearMotorDirection = DcMotorSimple.Direction.REVERSE
        FollowerConstants.rightFrontMotorDirection = DcMotorSimple.Direction.FORWARD
        FollowerConstants.rightRearMotorDirection = DcMotorSimple.Direction.FORWARD

        FollowerConstants.mass = TODO("Set robot mass")

        FollowerConstants.xMovement = 81.34056
        FollowerConstants.yMovement = 65.43028

        FollowerConstants.forwardZeroPowerAcceleration = -41.278
        FollowerConstants.lateralZeroPowerAcceleration = -59.7819

        FollowerConstants.translationalPIDFCoefficients.setCoefficients(0.1, 0.0, 0.0, 0.0)
        FollowerConstants.useSecondaryTranslationalPID = false

        FollowerConstants.headingPIDFCoefficients.setCoefficients(2.0, 0.0, 0.1, 0.0)
        FollowerConstants.useSecondaryHeadingPID = false

        FollowerConstants.drivePIDFCoefficients.setCoefficients(0.1, 0.0, 0.0, 0.6, 0.0)
        FollowerConstants.useSecondaryDrivePID = false

        FollowerConstants.zeroPowerAccelerationMultiplier = 4.0
        FollowerConstants.centripetalScaling = 0.0005

        FollowerConstants.pathEndTimeoutConstraint = 500.0
        FollowerConstants.pathEndTValueConstraint = 0.995
        FollowerConstants.pathEndVelocityConstraint = 0.1
        FollowerConstants.pathEndTranslationalConstraint = 0.1
        FollowerConstants.pathEndHeadingConstraint = 0.007
    }
}